

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>episcanpy.api.ct.build_count_mtx &mdash; epiScanpy 0.3.1+47.gb297b30 documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/typehints.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> epiScanpy
          

          
          </a>

          
            
            
              <div class="version">
                0.3.1+47.gb297b30
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../examples.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../basic_usage.html">Usage Principles</a></li>
<li class="toctree-l1"><a class="reference internal" href="../System_Requirements.html">System Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">epiScanpy</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          



















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>episcanpy.api.ct.build_count_mtx</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            
              <!-- User defined GitHub URL -->
              <a href="https://github.com/colomemaria/epiScanpy/tree/master/episcanpy/count_matrix/_bld_met_mtx.py#L431-L568" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="episcanpy-api-ct-build-count-mtx">
<h1>episcanpy.api.ct.build_count_mtx<a class="headerlink" href="#episcanpy-api-ct-build-count-mtx" title="Permalink to this headline">¶</a></h1>
<dl class="py function">
<dt class="sig sig-object py" id="episcanpy.api.ct.build_count_mtx">
<span class="sig-prename descclassname"><span class="pre">episcanpy.api.ct.</span></span><span class="sig-name descname"><span class="pre">build_count_mtx</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">cells</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">annotation</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">''</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">output_file</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">writing_option</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'w'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">meth_context</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'CG'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">chromosome</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">feature_names</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threshold</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">ct_mtx</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">sparse</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">copy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#episcanpy.api.ct.build_count_mtx" title="Permalink to this definition">¶</a></dt>
<dd><p>Build methylation count matrix for a given annotation.
It either write the count matrix (if given an output file) or return it as a variable (numpy matrix).
If you want to add cells to an already existing matrix (with the same annotations),  you put the initial matrix as ct_mtx or you specify the matrix to write +
writing option = a</p>
<p>if you want to write down the matrix as a sparse matrix you have to specify it (not implented yet)</p>
<p>I need to pay attention to where I am writing the output file.</p>
<p>Also, verbosity..</p>
<p>Pay attention, it does not average variables. If you want to process many small features such as
tfbs, we advise to use the dedicated function.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>cells</strong></dt><dd><p>list of the file names to read to build the count matrix.</p>
</dd>
<dt><strong>annotation</strong></dt><dd><p>loaded annotation to use to build the count matrix
‘str’ or ‘list’ depending of the number of matrices to build</p>
</dd>
<dt><strong>path</strong></dt><dd><p>path to the input data.</p>
</dd>
<dt><strong>output_file</strong></dt><dd><p>name files to write. ‘str’ or ‘list’ depending of the number of matrices to build</p>
</dd>
<dt><strong>writing_option</strong></dt><dd><p>either ‘w’ if you want to erase potentialy already existing file or ‘a’ to append.
‘str’ or ‘list’ if you have a list of matrices and the writing options are differents</p>
</dd>
<dt><strong>meth_context</strong></dt><dd><p>read methylation in ‘CG’ of ‘CH’ context</p>
</dd>
<dt><strong>chromosome</strong></dt><dd><p>‘MOUSE’ and ‘HUMAN’ (without mitochondrial genome) or list with chromosomes.
If None, chromosomes are deduced from the chromosomes containing features.</p>
</dd>
<dt><strong>feature_names</strong></dt><dd><p>If you want to write down the name of the annotation features.
‘Int’ (or ‘list’ if you have multiple annotations)</p>
</dd>
<dt><strong>threshold</strong></dt><dd><p>the minimum of cytosines covered per annotation to calculate a methylation level.
default=1 ‘Int’ (or ‘list’ if you have multiple annotations with different thresholds)</p>
</dd>
<dt><strong>sparse</strong></dt><dd><p>Boolean, writing option as a normal or sparse matrix.
default: False</p>
</dd>
<dt><strong>copy</strong></dt><dd><p>if True, return count matrix</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Anna Danese, Maria Richter, Kridsadakorn Chaichoompu..

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>