

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>episcanpy.api &mdash; epiScanpy 0.3.1+47.gb297b30 documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/typehints.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> epiScanpy
          

          
          </a>

          
            
            
              <div class="version">
                0.3.1+47.gb297b30
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="../examples.html">Tutorials</a></li>
<li class="toctree-l1"><a class="reference internal" href="../basic_usage.html">Usage Principles</a></li>
<li class="toctree-l1"><a class="reference internal" href="../System_Requirements.html">System Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="index.html">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../references.html">References</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">epiScanpy</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>episcanpy.api</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            
              <a href="https://github.com/colomemaria/epiScanpy/blob/master/docs/api/episcanpy.api.rst" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="episcanpy-api">
<h1>episcanpy.api<a class="headerlink" href="#episcanpy-api" title="Permalink to this headline">¶</a></h1>
<span class="target" id="module-episcanpy.api"></span><div class="section" id="api">
<h2>API<a class="headerlink" href="#api" title="Permalink to this headline">¶</a></h2>
<p>Import epiScanpy’s high-level API as:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">episcanpy.api</span> <span class="k">as</span> <span class="nn">epi</span>
</pre></div>
</div>
<div class="section" id="count-matrices-ct">
<h3>Count Matrices: CT<a class="headerlink" href="#count-matrices-ct" title="Permalink to this headline">¶</a></h3>
<p>Loading data, loading annotations, building count matrices, filtering of lowly covered methylation variables.
Filtering of lowly covered cells.</p>
<div class="section" id="load-features">
<h4>Load features<a class="headerlink" href="#load-features" title="Permalink to this headline">¶</a></h4>
<p>In order to build a count matrix for either methylation or open chromatin data, loading the segmentation of the genome of interest or the set of features of interest is a prerequirement.</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.ct.load_features.html#episcanpy.api.ct.load_features" title="episcanpy.api.ct.load_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.load_features</span></code></a>(file_features[, …])</p></td>
<td><p>The function load features is here to transform a bed file into a usable set of units to measure methylation levels.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.ct.make_windows.html#episcanpy.api.ct.make_windows" title="episcanpy.api.ct.make_windows"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.make_windows</span></code></a>(size[, chromosomes, …])</p></td>
<td><p>Generate windows/bins of the given size for the appropriate genome (default choice is human).</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.ct.size_feature_norm.html#episcanpy.api.ct.size_feature_norm" title="episcanpy.api.ct.size_feature_norm"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.size_feature_norm</span></code></a>(loaded_feature, size)</p></td>
<td><p>If the features loaded are too smalls or of different sizes, it is possible to normalise them to a unique given size by extending the feature coordinate in both directions.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.ct.plot_size_features.html#episcanpy.api.ct.plot_size_features" title="episcanpy.api.ct.plot_size_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.plot_size_features</span></code></a>(loaded_feature[, …])</p></td>
<td><p>Plot the different feature sizes in an histogram.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.ct.name_features.html#episcanpy.api.ct.name_features" title="episcanpy.api.ct.name_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.name_features</span></code></a>(loaded_features)</p></td>
<td><p>Extract the names of the loaded features, specifying the chromosome they originated from.</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="reading-methylation-file">
<h4>Reading methylation file<a class="headerlink" href="#reading-methylation-file" title="Permalink to this headline">¶</a></h4>
<p>Functions to read methylation files, extract methylation and buildthe count matrices:</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.ct.build_count_mtx.html#episcanpy.api.ct.build_count_mtx" title="episcanpy.api.ct.build_count_mtx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.build_count_mtx</span></code></a>(cells, annotation[, …])</p></td>
<td><p>Build methylation count matrix for a given annotation.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.ct.read_cyt_summary.html#episcanpy.api.ct.read_cyt_summary" title="episcanpy.api.ct.read_cyt_summary"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.read_cyt_summary</span></code></a>(sample_name, meth_type, …)</p></td>
<td><p>Read file from which you want to extract the methylation level and (assuming it is like the Ecker/Methylpy format) extract the number of methylated read and the total number of read for the cytosines covered and in the right genomic context (CG or CH) :param sample_name: name of the file to read to extract key information.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.ct.load_met_noimput.html#episcanpy.api.ct.load_met_noimput" title="episcanpy.api.ct.load_met_noimput"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.load_met_noimput</span></code></a>(matrix_file[, path, save])</p></td>
<td><p>read the raw count matrix and convert it into an AnnData object.</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="reading-open-chromatin-atac-file">
<h4>Reading open chromatin(ATAC) file<a class="headerlink" href="#reading-open-chromatin-atac-file" title="Permalink to this headline">¶</a></h4>
<p>ATAC-seq specific functions to build count matrices and load data:</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.ct.bld_atac_mtx.html#episcanpy.api.ct.bld_atac_mtx" title="episcanpy.api.ct.bld_atac_mtx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.bld_atac_mtx</span></code></a>(list_bam_files, loaded_feat)</p></td>
<td><p>Build a count matrix one set of features at a time.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.ct.save_sparse_mtx.html#episcanpy.api.ct.save_sparse_mtx" title="episcanpy.api.ct.save_sparse_mtx"><code class="xref py py-obj docutils literal notranslate"><span class="pre">ct.save_sparse_mtx</span></code></a>(initial_matrix[, …])</p></td>
<td><p>Convert regular atac matrix into a sparse Anndata:</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="general-functions">
<h4>General functions<a class="headerlink" href="#general-functions" title="Permalink to this headline">¶</a></h4>
<p>Functions non -omic specific:</p>
<blockquote>
<div><table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
</tbody>
</table>
<dl class="field-list simple">
<dt class="field-odd">toctree</dt>
<dd class="field-odd"><p>.</p>
</dd>
</dl>
<p>ct.save_sparse_mtx</p>
</div></blockquote>
</div>
</div>
<div class="section" id="preprocessing-pp">
<h3>Preprocessing: PP<a class="headerlink" href="#preprocessing-pp" title="Permalink to this headline">¶</a></h3>
<p>Imputing missing data (methylation), filtering lowly covered cells or variables, correction for batch effect.</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.pp.coverage_cells.html#episcanpy.api.pp.coverage_cells" title="episcanpy.api.pp.coverage_cells"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.coverage_cells</span></code></a>(adata[, key_added, log, …])</p></td>
<td><p>Histogram of the number of open features (in the case of ATAC-seq data) per cell.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.pp.binarize.html#episcanpy.api.pp.binarize" title="episcanpy.api.pp.binarize"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.binarize</span></code></a>(adata[, copy])</p></td>
<td><p>convert the count matrix into a binary matrix.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.pp.lazy.html#episcanpy.api.pp.lazy" title="episcanpy.api.pp.lazy"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.lazy</span></code></a>(adata[, pp_pca, svd_solver, nb_pcs, …])</p></td>
<td><p>Automatically computes PCA coordinates, loadings and variance decomposition, a neighborhood graph of observations, t-distributed stochastic neighborhood embedding (tSNE) Uniform Manifold Approximation and Projection (UMAP)</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.pp.load_metadata.html#episcanpy.api.pp.load_metadata" title="episcanpy.api.pp.load_metadata"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.load_metadata</span></code></a>(adata, metadata_file[, …])</p></td>
<td><p>Load observational metadata in adata.obs.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.pp.read_ATAC_10x.html#episcanpy.api.pp.read_ATAC_10x" title="episcanpy.api.pp.read_ATAC_10x"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.read_ATAC_10x</span></code></a>(matrix[, cell_names, …])</p></td>
<td><p>Load sparse matrix (including matrices corresponding to 10x data) as AnnData objects.</p></td>
</tr>
</tbody>
</table>
<div class="section" id="methylation-matrices">
<h4>Methylation matrices<a class="headerlink" href="#methylation-matrices" title="Permalink to this headline">¶</a></h4>
<p>Methylation specific count matrices.</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.pp.imputation_met.html#episcanpy.api.pp.imputation_met" title="episcanpy.api.pp.imputation_met"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.imputation_met</span></code></a>(adata[, …])</p></td>
<td><p>Impute missing values in methyaltion level matrices.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.pp.load_met_noimput.html#episcanpy.api.pp.load_met_noimput" title="episcanpy.api.pp.load_met_noimput"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.load_met_noimput</span></code></a>(matrix_file[, path, save])</p></td>
<td><p>read the raw count matrix and convert it into an AnnData object.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.pp.readandimputematrix.html#episcanpy.api.pp.readandimputematrix" title="episcanpy.api.pp.readandimputematrix"><code class="xref py py-obj docutils literal notranslate"><span class="pre">pp.readandimputematrix</span></code></a>(file_name[, min_coverage])</p></td>
<td><p>Temporary function to load and impute methyaltion count matrix into an AnnData object</p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="tools-tl">
<h3>Tools: TL<a class="headerlink" href="#tools-tl" title="Permalink to this headline">¶</a></h3>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.tl.rank_features.html#episcanpy.api.tl.rank_features" title="episcanpy.api.tl.rank_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">tl.rank_features</span></code></a>(adata, groupby[, omic, …])</p></td>
<td><p>It is a wrap-up function of scanpy sc.tl.rank_genes_groups function.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.tl.silhouette.html#episcanpy.api.tl.silhouette" title="episcanpy.api.tl.silhouette"><code class="xref py py-obj docutils literal notranslate"><span class="pre">tl.silhouette</span></code></a>(adata_name, cluster_annot[, …])</p></td>
<td><p>Compute silhouette scores.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.tl.lazy.html#episcanpy.api.tl.lazy" title="episcanpy.api.tl.lazy"><code class="xref py py-obj docutils literal notranslate"><span class="pre">tl.lazy</span></code></a>(adata[, pp_pca, copy])</p></td>
<td><p>Automatically computes PCA coordinates, loadings and variance decomposition, a neighborhood graph of observations, t-distributed stochastic neighborhood embedding (tSNE) Uniform Manifold Approximation and Projection (UMAP)</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="episcanpy.api.tl.load_markers.html#episcanpy.api.tl.load_markers" title="episcanpy.api.tl.load_markers"><code class="xref py py-obj docutils literal notranslate"><span class="pre">tl.load_markers</span></code></a>(path, marker_list_file)</p></td>
<td><p>Convert list of known cell type markers from literature to a dictionary Input list of known marker genes First row is considered the header</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="episcanpy.api.tl.identify_cluster.html#episcanpy.api.tl.identify_cluster" title="episcanpy.api.tl.identify_cluster"><code class="xref py py-obj docutils literal notranslate"><span class="pre">tl.identify_cluster</span></code></a>(adata, cell_type, …[, …])</p></td>
<td><p>Use markers of a given cell type to plot peak openness for peaks in promoters of the given markers Input cell type, cell type markers, peak promoter intersections</p></td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="plotting-pl">
<h3>Plotting: PL<a class="headerlink" href="#plotting-pl" title="Permalink to this headline">¶</a></h3>
<p>The plotting module <code class="xref py py-class docutils literal notranslate"><span class="pre">episcanpy.plotting</span></code> largely parallels the <code class="docutils literal notranslate"><span class="pre">tl.*</span></code> and a few of the <code class="docutils literal notranslate"><span class="pre">pp.*</span></code> functions.
For most tools and for some preprocessing functions, you’ll find a plotting function with the same name.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Anna Danese, Maria Richter, Kridsadakorn Chaichoompu..

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>